Merge apis from unstable

vividviolet opened the pull request:
### Background
Fix https://github.com/shop/issues-appui/issues/196
Cherry-pick commits from `unstable` for Admin that are missing from this branch

### Tophat

2. Create an Admin Action extension using `POLARIS_UNIFIED=true`
3. Update the package.json to pull in the snapshot
4. Verify that `shopify.picker` and `shopify.resourcePicker` are both available
   <img width="1139" alt="image" src="https://github.com/user-attachments/assets/516bda85-6751-4cc3-abaf-414e1b0c4f2f" />

MitchLillie committed:
Add generic picker API

JulienHe committed:
Add initial target

JulienHe committed:
Add screenshots

JulienHe committed:
Change the image for the rendering

charlesdobson committed:
expose picker only to block extensions and add static doc section

melissaluu committed:
Add resourcePicker to ActionExtensionApi interface

charlesdobson committed:
update picker type to support new undefined resolution on cancel

charlesdobson committed:
add doc blocks for picker types

aeperea committed:
Defaulting to string type for enabled features passed to the templates

MitchLillie committed:
Nest target APIs into a subsection

MitchLillie committed:
Sync examples and typedocs with AB

devisscher committed:
Add discount functions settings api docs

MitchLillie committed:
Update picker links to ui-ext picker

melissaluu committed:
Update PrintActionExtensionApii to include ResourcePicker

MitchLillie committed:
Add tips for resource picker and picker

charlesdobson committed:
update docs copy

Co-authored-by: Jason Miller <developit@users.noreply.github.com>

melissaluu committed:
Add picker references and update picker docs

kmdavis committed:
feat(products): updating Product Configuration extension points to use block extension api & components

jonathanhamel4 committed:
use the block api again for the discount settings api

vividviolet committed:
Fix purchase options api types

vividviolet committed:
Edit changeset to remove deprecated packages

vividviolet committed:
Fix Badge types for picker

vividviolet commented:
/snapit

Fionoble commented on packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:
```diff
@@ -38,10 +38,34 @@ const data: LandingTemplateSchema = {
+          subtitle: 'App authentication',
+          name: "Make authenticated requests to your app's backend",
+          url: '#app-authentication',
+          type: 'tool',
```
Direct api, app auth and custom protocols should stay under network features.

Fionoble reviewed (changes requested):
Just the one revision, otherwise this should be good.

vividviolet committed:
Remove duplicated network api docs
Rename Picker api's internal Badge interface so it does not collide with the real Badge component

Fionoble commented on packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:
```diff
@@ -38,10 +38,16 @@ const data: LandingTemplateSchema = {
-          subtitle: 'Network Features',
-          name: 'Learn about the network features available to admin extensions',
-          url: '/docs/api/admin-extensions/api/network-features',
-          type: 'globe',
```
Can you add this card back? Should be good to go after that (sorry)

vividviolet replied:
Oops, my bad
